src/spatial_ref/srs.rs

```diff
@@ -203,14 +203,16 @@ impl SpatialRef {
     pub fn to_wkt(&self) -> Result<String> {
         let mut c_wkt = ptr::null_mut();
         let rv = unsafe { gdal_sys::OSRExportToWkt(self.0, &mut c_wkt) };
-        if rv != OGRErr::OGRERR_NONE {
+        let res = if rv != OGRErr::OGRERR_NONE {
             Err(GdalError::OgrError {
                 err: rv,
                 method_name: "OSRExportToWkt",
             })
         } else {
             Ok(_string(c_wkt))
-        }
+        };
+        unsafe { gdal_sys::VSIFree(c_wkt.cast::<std::ffi::c_void>()) };
```

Comment:
For all `c_wkt` in this file. What is the pointer if we get an error? Do we need to destroy it or is this a null pointer?

Reply:
with the changes we always call free on the pointer

Reply:
Yes, but I can't find the info on VSIFree that it is okay to insert a null pointer.
If this is the case, we can resolve this.

Reply:
it is ok to call free on nullotr  in c: https://linux.die.net/man/3/free

and i think it is ok to call VSIfree on nullptr:
https://github.com/OSGeo/gdal/blob/fec15b146f8a750c23c5e765cac12ed5fc9c2b85/gdal/port/cpl_vsisimple.cpp#L827

Reply:
Ok for me, still I can't resolve this conversation.